modules/widgets/migrate-to-core/image-widget.php

```diff
@@ -0,0 +1,119 @@
+		'width' => 0,
+		'height' => 0,
+		'align' => 'none',
+		'caption' => '',
+		'alt' => '',
+		'link_type' => '',
+		'link_url' => '',
+		'image_classes' => '',
+		'link_classes' => '',
+		'link_rel' => '',
+		'image_title' => '',
+		'link_target_blank' => false,
+	);
+
+	$media_image      = get_option( 'widget_media_image' );
+	$sidebars_widgets = wp_get_sidebars_widgets();
+
+	foreach ( get_option( 'widget_image' ) as $id => $widget ) {
+		if ( is_string( $id ) ) {
+			continue;
+		}
+
+		$media_image[ $id ] = array_merge( $default_data, array_intersect_key( $widget, $default_data ), array(
+			'alt'         => $widget['alt_text'],
+			'height'      => $widget['img_height'],
+			'image_title' => $widget['img_title'],
+			'link_url'    => $widget['link'],
+			'url'         => $widget['img_url'],
+			'width'       => $widget['img_width'],
+		) );
+
+		// Check if the image is in the media library.
+		$image_basename = basename( $widget['img_url'] );
+		$attachment_ids = get_posts( array(
+			'fields'      => 'ids',
+			'meta_query'  => array(
+				array(
+					'value'   => basename( $image_basename ),
+					'compare' => 'LIKE',
+					'key'     => '_wp_attachment_metadata',
+				),
+			),
+			'post_status' => 'inherit',
+			'post_type'   => 'attachment',
+		) );
+
+		foreach ( (array) $attachment_ids as $attachment_id ) {
+			$image_meta = wp_get_attachment_metadata( $attachment_id );
+
+			// Is it a full size image?
+			if ( $image_basename === array_pop( explode( '/', $image_meta['file'] ) ) ) {
+				$media_image[ $id ]['attachment_id'] = $attachment_id;
+				$media_image[ $id ]['width']         = $image_meta['width'];
+				$media_image[ $id ]['height']        = $image_meta['height'];
+				break;
+			}
+
+			// Is it a down-sized image?
+			foreach ( $image_meta['sizes'] as $size => $image ) {
+				if ( false !== array_search( $image_basename, $image ) ) {
+					$media_image[ $id ]['attachment_id'] = $attachment_id;
+					$media_image[ $id ]['size']          = $size;
+					$media_image[ $id ]['width']         = $image['width'];
+					$media_image[ $id ]['height']        = $image['height'];
+					break 2;
+				}
+			}
+		}
+
+		if ( ! empty( $widget['link'] ) ) {
+			$media_image[ $id ]['link_type'] = $widget['link'] === $widget['img_url'] ? 'file' : 'custom';
+		}
+
+		foreach ( $sidebars_widgets as $sidebar => $widgets ) {
+			if ( false !== ( $key = array_search( "image-{$id}", $widgets, true ) ) ) {
+				$sidebars_widgets[ $sidebar ][ $key ] = "media_image-{$id}";
+			}
+		}
+
+		wp_unregister_sidebar_widget( "image-{$id}" );
+		$media_image_widget = new WP_Widget_Media_Image();
+		$media_image_widget->_set( $id );
+		$media_image_widget->_register_one( $id );
+	}
+
+	update_option( 'widget_media_image', $media_image );
+	delete_option( 'widget_image' );
+	wp_set_sidebars_widgets( $sidebars_widgets );
+
+	Jetpack_Options::update_option( 'image_widget_migration', true );
+}
+add_action( 'widgets_init', 'jetpack_migrate_image_widget' );
```

Comment:
Maybe this could be called in `Jetpack::plugin_upgrade` https://github.com/Automattic/jetpack/blob/master/class.jetpack.php#L335
that runs whenever a different version is detected so it's run only once and isn't always hooked to `widgets_init` and loading the option to check?